### PR TITLE
Code Readability - Game (misc)

### PR DESCRIPTION
**Pull request by KalevTait**
## What Does This PR Do
* Converts 0/1 to FALSE/TRUE where those #defines make the code more readable.
* Converts some stat_allowed = 0 to stat_allowed = CONSCIOUS
* Where bools are in if statements, uses them as bools
* For mecha equipment, converts returns that should be TRUE or FALSE to those values
* For mecha equipment with states, gets rid of magic numbers by using #defines to specify state

## Why It's Good For The Game
More readable code is easier to work with.

### CODE REVIEW

File: code/game/mecha/equipment/weapons/weapons.dm [hunk 1]
```diff
@@ -18,7 +18,7 @@
 			if(size > M.maxsize)
 				return 0
```
Comment (Sirryan2002) [inline code/game/mecha/equipment/weapons/weapons.dm:19]:
```suggestion
				return
```

File: code/game/mecha/equipment/weapons/weapons.dm [hunk 2]
```diff
@@ -18,7 +18,7 @@
...
 			return 1
```
Comment (Sirryan2002) [inline code/game/mecha/equipment/weapons/weapons.dm:20]:
```suggestion
			return TRUE
```

File: code/game/mecha/equipment/weapons/weapons.dm [hunk 3]
```diff
@@ -18,7 +18,7 @@
...
-		else if(M.emagged == 1)
+		else if(M.emagged)
 			return 1
```
Comment (Sirryan2002) [inline code/game/mecha/equipment/weapons/weapons.dm:22]:
```suggestion
			return TRUE
```

File: code/game/mecha/equipment/weapons/weapons.dm [hunk 4]
```diff
@@ -18,7 +18,7 @@
...
 	return 0
```
Comment (Sirryan2002) [inline code/game/mecha/equipment/weapons/weapons.dm:23]:
```suggestion
	return
```

File: code/game/mecha/equipment/weapons/weapons.dm (hunk 1 above)
Comment (KalevTait) [reply code/game/mecha/equipment/weapons/weapons.dm]: Done, plus a bit more....

File: code/game/mecha/equipment/tools/other_tools.dm [hunk 5]
```diff
@@ -85,7 +88,7 @@
 			var/turf/T = get_turf(target)
 			cooldown_timer = world.time + 3 SECONDS
 			log_game("[key_name(chassis.occupant)] used a Gravitational Catapult in ([T.x],[T.y],[T.z])")
-			return 1
+			return
```
Comment (hal9000PR) [inline code/game/mecha/equipment/tools/other_tools.dm:91]: dangling return, (a return value at the end of a proc that returns nothing) is unneeded.

File: code/game/mecha/equipment/tools/other_tools.dm [hunk 6]
```diff
@@ -1,6 +1,9 @@
 // Teleporter, Gravitational catapult, Armor booster modules,
 // Repair droid, Tesla Energy relay, Generators
 
+#define MECH_GRAVCAT_MODE_GRAVSLING 1
+#define MECH_GRAVCAT_MODE_GRAVPUSH 2
+
```
Comment (hal9000PR) [inline code/game/mecha/equipment/tools/other_tools.dm:6]: `#undef` these at the end of the file.

File: code/game/mecha/equipment/tools/work_tools.dm [hunk 7]
```diff
@@ -1,5 +1,10 @@
+
 //Hydraulic clamp, Kill clamp, Extinguisher, RCD, Mime RCD, Cable layer.
 
+#define MECH_RCD_MODE_DECONSTRUCT 0
+#define MECH_RCD_MODE_WALL_OR_FLOOR 1
+#define MECH_RCD_MODE_AIRLOCK 2
+
```
Comment (hal9000PR) [inline code/game/mecha/equipment/tools/work_tools.dm:7]: undef these

File: code/game/mecha/equipment/tools/work_tools.dm [hunk 8]
```diff
@@ -92,11 +97,11 @@
 		if(!O.anchored)
 			if(cargo_holder.cargo.len < cargo_holder.cargo_capacity)
 				chassis.visible_message("[chassis] lifts [target] and starts to load it into cargo compartment.")
-				O.anchored = 1
+				O.anchored = TRUE
 				if(do_after_cooldown(target))
 					cargo_holder.cargo += O
 					O.loc = chassis
```
Comment (hal9000PR) [inline code/game/mecha/equipment/tools/work_tools.dm:103]:
```suggestion
					O.forceMove(chassis)
```

File: code/game/mecha/mecha_construction_paths.dm [hunk 9]
```diff
@@ -551,7 +551,7 @@
 	const_holder.construct = new /datum/construction/reversible/mecha/firefighter(const_holder)
 	const_holder.icon = 'icons/mecha/mech_construction.dmi'
 	const_holder.icon_state = "fireripley0"
-	const_holder.density = 1
+	const_holder.density = TRUE
 	qdel(src)
 	return
```
Comment (hal9000PR) [inline code/game/mecha/mecha_construction_paths.dm:556]: lots of dangling returns in this file. please remove.

**Review by hal9000PR — CHANGES_REQUESTED** [review]

**Review by hal9000PR — APPROVED** [review]

**Pull request merged**